src/wallet/rpcwallet.cpp

```diff
@@ -243,16 +254,24 @@ UniValue getrawchangeaddress(const JSONRPCRequest& request)
         pwallet->TopUpKeyPool();
     }
 
+    OutputType output_type = g_change_type;
+    if (!request.params[0].isNull()) {
+        output_type = ParseOutputType(request.params[0].get_str());
```

Comment:
I believe this allows the user to put in "default", which then resolves to "P2SH" no matter what the actual default g_change_type is set to .